jax/core.py

```diff
@@ -639,21 +639,38 @@ def __repr__(self): return '*'
 identity_p.def_impl(lambda x: x)
 identity_p.def_custom_bind(lambda x: x)
 
-def concretization_err_msg(fun, context=None):
+class ConcretizationTypeError(TypeError): pass
+
+def raise_concretization_error(val, context=""):
+  msg = (f"Abstract tracer value encountered where a concrete value is expected ({context}).\n"
```

Comment:
How about: "Tracer value (at an abstraction level higher than concrete) encountered where a concrete value is expected."?